_examples/custom_http_auth/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"gopkg.in/src-d/go-git.v4"
+	. "gopkg.in/src-d/go-git.v4/_examples"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/client"
+	githttp "gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+// Here is an example to configure http client according to our own needs.
+func main() {
+	CheckArgs("<url>", "<username>", "<password>")
+	url := os.Args[1]
+	username := os.Args[2]
+	password := os.Args[3]
+
+	// Create a custom http(s) client with your config
+	customClient := &http.Client{
+		// accept any certificate (might be useful for testing)
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+
+		// 15 second timeout
+		Timeout: 15 * time.Second,
+
+		// don't follow redirect
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	// Override http(s) default protocol to use our custom client
+	client.InstallProtocol("https", githttp.NewClient(customClient))
```

Comment:
Apologies in advance if this is merely my own ignorance, but is there a specific reason we need to include this in the example code? It seems to me like encouraging people to disable SSL verification and redirects isn't good general practice. I think this is more or less copy/pasted from the existing `custom_http` example but maybe doesn't make sense here.

For this example code I think it makes more sense to just focus on demonstrating the basic http auth usage in isolation.